src/coxpcall.lua

```diff
@@ -40,18 +48,91 @@ local function id(trace)
 	return trace
 end
 
+local function filter_outer_tb(tb)
+	if type(tb) ~= 'string' or tb == '' then
+		return nil
+	end
+
+	local kept = {}
+	for line in tb:gmatch('[^\n]+') do
+		if line ~= 'stack traceback:'
+			and not line:match('^%s*$')
+			and not line:match('%(tail call%)')
+			and not line:find('coxpcall.lua', 1, true)
+			and not line:find("in function 'coroutine.resume'", 1, true)
+			and not line:find('handleReturnValue', 1, true)
+			and not line:find('performResume', 1, true)
+		then
+			kept[#kept + 1] = line
+		end
+	end
+
+	if #kept == 0 then
+		return nil
+	end
+	return table.concat(kept, '\n')
+end
+
+local function splice_chain(tb_inner, co, marker)
+	if type(tb_inner) ~= 'string' or tb_inner == '' then
+		return tb_inner
+	end
+	if not (debug and debug.traceback) then
+		return tb_inner
+	end
+
+	marker = marker or "\t[C]: in function 'xpcall'"
+
+	local out = tb_inner
+	local parent = coromap[co]
```

Comment:
Could a bug introduce an infinite parent loop here?  Should we guard against that?